src/components/select/multi-select/multi-select.component.tsx

```diff
@@ -149,27 +149,24 @@ export const MultiSelect = React.forwardRef<HTMLInputElement, MultiSelectProps>(
     const accessibilityLabelId = useRef(guid());
     const containerRef = useRef<HTMLDivElement>(null);
     const listboxRef = useRef<HTMLDivElement>(null);
-    const isInputFocused = useRef(false);
     const isClickTriggeredBySelect = useRef(false);
     const isMouseDownReported = useRef(false);
     const isMouseDownOnInput = useRef(false);
-    const isOpenedByFocus = useRef(false);
+    const [isOpenedByFocus, setIsOpenedByFocus] = useState(false);
```

Comment:
**question(non-blocking):** What has been the thinking in moving this from `ref` to `state`? Did you discover an issue or something else when writing the new tests?

Reply:
I think the changes have improved the code, but I'm not convinced we needed to do this at this time as we are just trying to get this refactored to RTL and all the Selects are being rewritten anyway

Reply:
I updated this to state after identifying it as a reactive value. It's generally recommended to [avoid writing to refs during rendering](https://react.dev/learn/referencing-values-with-refs#best-practices-for-refs) as it can make the component's behaviour more difficult to reason.

While this change isn't strictly necessary for the RTL refactor, I chose to make it during refactoring as it also helped simplify the component's code, which I decided to do to achieve 100% test coverage. Although modifying a component just to pass tests doesn't seem ideal, I ran into challenges with RTL's user-centric approach. Without these changes, I was concerned about introducing excessive bloat to the tests. Given that a full component rewrite is unlikely soon, I opted to make small, incremental improvements where possible.